assets/alertmanager-user-workload/alertmanager.yaml

```diff
@@ -167,7 +167,16 @@ spec:
       type: RuntimeDefault
   serviceAccountName: alertmanager-user-workload
   version: 0.26.0
+  volumeMounts:
+  - mountPath: /etc/pki/ca-trust/extracted/pem/
+    name: alertmanager-trusted-ca-bundle
   volumes:
   - configMap:
       name: metrics-client-ca
     name: metrics-client-ca
+  - configMap:
+      items:
+      - key: ca-bundle.crt
+        path: tls-ca-bundle.pem
+      name: alertmanager-trusted-ca-bundle
```

Comment:
should the name be like `alertmanager-trusted-ca-bundle-2ua4n9ob5qr8o`
tested with this PR
`launch 4.16.0-0.nightly-2024-05-16-092402,openshift/cluster-monitoring-operator#2349 aws,no-spot`

```
$ oc -n openshift-user-workload-monitoring get cm | grep alertmanager-trusted-ca-bundle
alertmanager-trusted-ca-bundle                             1      3m49s
alertmanager-trusted-ca-bundle-2ua4n9ob5qr8o               1      3m44s
```
alertmanager-trusted-ca-bundle-2ua4n9ob5qr8o cofigmap is created but not referenced in  pod or sts
```
$ oc -n openshift-user-workload-monitoring get sts alertmanager-user-workload -oyaml | grep alertmanager-trusted-ca-bundle -C3
          name: secret-alertmanager-kube-rbac-proxy-metric
          readOnly: true
        - mountPath: /etc/pki/ca-trust/extracted/pem/
          name: alertmanager-trusted-ca-bundle
        - mountPath: /etc/alertmanager/web_config/web-config.yaml
          name: web-config
          readOnly: true
--
          items:
          - key: ca-bundle.crt
            path: tls-ca-bundle.pem
          name: alertmanager-trusted-ca-bundle
        name: alertmanager-trusted-ca-bundle
  updateStrategy:
    type: RollingUpdate

$ oc -n openshift-user-workload-monitoring get pod alertmanager-user-workload-0 -oyaml | grep alertmanager-trusted-ca-bundle -C3
      name: secret-alertmanager-kube-rbac-proxy-metric
      readOnly: true
    - mountPath: /etc/pki/ca-trust/extracted/pem/
      name: alertmanager-trusted-ca-bundle
    - mountPath: /etc/alertmanager/web_config/web-config.yaml
      name: web-config
      readOnly: true
--
      items:
      - key: ca-bundle.crt
        path: tls-ca-bundle.pem
      name: alertmanager-trusted-ca-bundle
    name: alertmanager-trusted-ca-bundle
  - name: kube-api-access-qjlp5
    projected:
      defaultMode: 420

$ oc -n openshift-user-workload-monitoring get pod alertmanager-user-workload-0 -oyaml | grep  alertmanager-trusted-ca-bundle-2ua4n9ob5qr8o
no result

$ oc -n openshift-user-workload-monitoring get sts alertmanager-user-workload -oyaml | grep alertmanager-trusted-ca-bundle-2ua4n9ob5qr8o
no result
```

Reply:
FYI
```
$ oc -n openshift-user-workload-monitoring get cm | grep prometheus-user-workload-trusted-ca-bundle
prometheus-user-workload-trusted-ca-bundle                 1      23m
prometheus-user-workload-trusted-ca-bundle-2ua4n9ob5qr8o   1      23m

$ oc -n openshift-user-workload-monitoring get sts  prometheus-user-workload -oyaml | grep prometheus-user-workload-trusted-ca-bundle -C3
        terminationMessagePolicy: FallbackToLogsOnError
        volumeMounts:
        - mountPath: /etc/pki/ca-trust/extracted/pem/
          name: prometheus-user-workload-trusted-ca-bundle
          readOnly: true
        - mountPath: /etc/prometheus/config_out
          name: config-out
--
          items:
          - key: ca-bundle.crt
            path: tls-ca-bundle.pem
          name: prometheus-user-workload-trusted-ca-bundle-2ua4n9ob5qr8o
          optional: true
        name: prometheus-user-workload-trusted-ca-bundle
      - name: secret-grpc-tls
        secret:
          defaultMode: 420
```

Reply:
The `*--trusted-ca-bundle-xxx` configmaps were needed when we had oauth-proxy because it couldn't dynamically reload the CA, unlike Prometheus and Alertmanager. 
I'm cleaning up these old references in https://issues.redhat.com/browse/MON-3701.